pkg/image/importer/importer.go

```diff
@@ -282,6 +282,27 @@ func applyErrorToRepository(repository *importRepository, err error) {
 	}
 }
 
+func calculateImageSize(ctx gocontext.Context, repo distribution.Repository, image *api.Image) error {
+	bs := repo.Blobs(ctx)
+
+	layerSet := sets.NewString()
+	size := int64(0)
+	for i := range image.DockerImageLayers {
+		layer := &image.DockerImageLayers[i]
+		desc, err := bs.Stat(ctx, digest.Digest(layer.Name))
```

Comment:
I think that's acceptable.  I don't think this issue is as severe as
originally stated.  The cost of doing all the stats outweighs the value,
especially if this is going to be less of a problem soon.

So in general, the only thing we need to fix right now is the tests, open a
follow up issue, and handle it there.

On Wed, Jun 1, 2016 at 3:01 PM, Alexey Gladkov <notifications@github.com>
wrote:

> In pkg/image/importer/importer.go
> <https://github.com/openshift/origin/pull/9115#discussion_r65422272>:
>
> > @@ -282,6 +282,27 @@ func applyErrorToRepository(repository *importRepository, err error) {
> >  	}
> >  }
> >
> > +func calculateImageSize(ctx gocontext.Context, repo distribution.Repository, image *api.Image) error {
> > +	bs := repo.Blobs(ctx)
> > +
> > +	layerSet := sets.NewString()
> > +	size := int64(0)
> > +	for i := range image.DockerImageLayers {
> > +		layer := &image.DockerImageLayers[i]
> > +		desc, err := bs.Stat(ctx, digest.Digest(layer.Name))
>
> Yes, but i'm talking about importer and hub.docker.com. The server can
> return schema1 to use even if we support the scheme2.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9115/files/492baa03f23cf45ca229c3501c29ca4bccf546f3#r65422272>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p5HLpG9t04zWq7V02kvPrZjHsCPdks5qHdcFgaJpZM4IrzC2>
> .
>